src/commands/ban_commands.py

```diff
@@ -15,30 +15,46 @@ def create_ban_commands(bot: Bot) -> None:
     @discord.app_commands.describe(
         user="User being banned",
         reason="Reason for ban",
+        delete_messages="Type 'Yes' to delete messages or 'No' to keep them (default: 'No')",
     )
     async def ban(
         interaction: discord.Interaction,
         user: discord.Member,
         reason: str,
+        delete_messages: str = "No",  # Default to 'No' for no message deletion
     ):
         """Ban the specified user."""
+        # Normalize and validate the input
+        delete_messages = delete_messages.strip().capitalize()  # Handle capitalization
+        if delete_messages not in ["Yes", "No"]:
+            await interaction.response.send_message(
+                "Invalid value for 'delete_messages'. Please enter 'Yes' to delete messages or 'No' to keep them.",
+                ephemeral=True,
+            )
+            return
+
+        # Convert delete_messages to a boolean for ban logic
+        delete_messages_flag = delete_messages == "Yes"
+
         async with create_response_context(interaction) as response_message:
-            (is_banned, is_dm_sent, result_description) = await ban_user(user, reason)
-
-            if is_banned:  # ban succeeded
-                if not is_dm_sent:  # dm failed
-                    async with create_logging_embed(
-                        interaction, user=user, reason=reason, error=result_description
-                    ) as logging_embed:
-                        response_message.set_string(result_description)
-                else:  # ban succeeded, dm failed.
-                    async with create_logging_embed(
-                        interaction, user=user, reason=reason, result=result_description
-                    ) as logging_embed:
-                        response_message.set_string(result_description)
-            else:  # ban failed, dont create embed
-                response_message.set_string(result_description)
-        response_message.set_string(result_description)
+            (is_banned, is_dm_sent, result_description) = await ban_user(
```

Comment:
since we moved `is_dm_sent` into the ban, we don't really need this out here (aka it's not being used in this function anymore). let's change the function to only return `is_banned` and `result_description`

Reply:
done